processor/src/test/resources/fixtures/org/mapstruct/ap/test/nestedbeans/mixed/FishTankMapperImpl.java

```diff
@@ -0,0 +1,358 @@
+/**
+ *  Copyright 2012-2017 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.nestedbeans.mixed;
+
+import javax.annotation.Generated;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.FishDto;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.FishTankDto;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.MaterialDto;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.MaterialTypeDto;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.OrnamentDto;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.WaterPlantDto;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.WaterQualityDto;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.WaterQualityOrganisationDto;
+import org.mapstruct.ap.test.nestedbeans.mixed._target.WaterQualityReportDto;
+import org.mapstruct.ap.test.nestedbeans.mixed.source.Fish;
+import org.mapstruct.ap.test.nestedbeans.mixed.source.FishTank;
+import org.mapstruct.ap.test.nestedbeans.mixed.source.Interior;
+import org.mapstruct.ap.test.nestedbeans.mixed.source.MaterialType;
+import org.mapstruct.ap.test.nestedbeans.mixed.source.Ornament;
+import org.mapstruct.ap.test.nestedbeans.mixed.source.WaterPlant;
+import org.mapstruct.ap.test.nestedbeans.mixed.source.WaterQuality;
+import org.mapstruct.ap.test.nestedbeans.mixed.source.WaterQualityReport;
+
+@Generated(
+    value = "org.mapstruct.ap.MappingProcessor",
+    date = "2017-02-12T21:48:04+0100",
+    comments = "version: , compiler: javac, environment: Java 1.8.0_45 (Oracle Corporation)"
+)
+public class FishTankMapperImpl implements FishTankMapper {
+
+    @Override
+    public FishTankDto map(FishTank source) {
+        if ( source == null ) {
+            return null;
+        }
+
+        FishTankDto fishTankDto = new FishTankDto();
+
+        fishTankDto.setMaterial( fishTankToMaterialDto( source ) );
```

Comment:
The reason for this are their mappings:

```java
@Mapping(target = "fish.kind", source = "fish.type"),
@Mapping(target = "material.materialType", source = "material"),
```

The first mapping is symmetric, which means that we allow forged name based mapping between `fishTankDto.getFish()` and `fishTank.getFish()`.

The second mapping is assymetric, and therefore we send the entire source object in the mapping. However this mapping does not forge name based mappings, it is only restricted to the defined mappings.  The extracted mapping for this will look like:

```java
@Mapping(target = "materialType", source = "material")
```